data/transactions/logic/jsonspec.md

```diff
@@ -136,3 +132,22 @@ replaced by U+FFFD
 ```json
 {"key0": "\uD800\uD800n"}
 ```
+
+### Keys
+
+- keys can be strings or numbers
+- strings must be quoted
+- number keys behave as quoted strings
```

Comment:
Interestingly, scientific notation and decimals are supported. This led me to think that maybe the Algorand codec doesn't care about quotes altogether but I have found that not to be the case. I'll be uploading my test cases shortly.

Reply:
Can this be turned off?  I think there's a strict mode. Maybe we need to use `codec` directly.

Reply:
I'd definitely prefer that we don't support numeric keys. As far as I'm aware, this feature is in our encoder because msgpack allows it, not because it's generally a nice thing to have for JSON.

Reply:
I spoke a bit with JJ about this. I could not find a way after taking a deep dive into the `go-codec` codebase this morning. It does seem not too difficult to add if we wanted to make a change to `go-codec`, but I was unable to find any existing way to exclude numeric keys.

Reply:
Very smart. I'll see what I can do with that.

Reply:
I was able to use the method Jason proposed to prohibit numeric keys 🎉